src/ros_filter.cpp

```diff
@@ -1981,10 +1981,7 @@ void RosFilter<T>::initialize()
   }
 
   const std::chrono::duration<double> timespan{1.0 / frequency_};
-  timer_ = rclcpp::GenericTimer<rclcpp::VoidCallbackType>::make_shared(
-    this->get_clock(), std::chrono::duration_cast<std::chrono::nanoseconds>(timespan),
-    std::bind(&RosFilter<T>::periodicUpdate, this), this->get_node_base_interface()->get_context());
-  this->get_node_timers_interface()->add_timer(timer_, nullptr);
+  timer_ = this->create_wall_timer(timespan, std::bind(&RosFilter<T>::periodicUpdate, this));
```

Comment:
I haven't used this package in earnest in ROS2, but if you are trying to run the EKF live against a bag file, and you pause it, won't this cause the EKF to continue to update?

Reply:
Yes, when I pause simulation EKF update method is still calling. Is this wrong?

Reply:
I am not really sure how this will behave offhand. What happens to the output topic when this happens?

Reply:
Still publishing at requested frequency with constant values

Reply:
OK, so I think the problem here will be this: even if you put this change in and left the Gazebo clock at 10 Hz, the other parts of the EKF that use the clock will (and must) continue to use ROS time for updating the filter. I'm assuming the internal ROS callback timing is also based on ROS time, but I could be wrong. If I'm not, though, your state estimate itself is still only going to actually update at 10 Hz, even if you tell the EKF to publish at 50 Hz. You'll get ~5 messages from the EKF with the same data in it, and then it will change and repeat for 5 more, etc.

In the end, I think the solution here is to make sure your Gazebo clock update rate is >= your EKF frequency. Feel free to add a note to the documentation somewhere with a note about this.